src/provider/model/ContextChannel.ts

```diff
@@ -81,7 +80,7 @@ export class DesktopContextChannel extends ContextChannelBase {
         this._context = null;
     }
 
-    public serialize(): DesktopChannelTransport {
-        return {name: this.name, color: this.color, ...super.serialize() as {id: string, type: 'desktop'}};
+    public serialize(): SystemChannelTransport {
+        return {visualIdentity: {...this.visualIdentity}, ...super.serialize() as {id: string, type: 'system'}};
```

Comment:
Do we need the clone on `visualIdentity`? I think the only thing we ever do with these serialised objects is immediately send them over the wire.

Should the return types of all these `serialize` functions be `Readonly<T>` types, so we know we can safely return without copying?